### ISSUE TITLE
Allow dashboard to be filtered by typing in it

### ISSUE THREAD

**Issue by pvginkel** [gitextensions/gitextensions#6641]
## Feature description

The dashboard tends to become unmanageable when you have a lot of projects, especially when you switch between active projects a lot. I often end up just using `Ctrl+O` and typing the project directory instead because it's faster.

It'd be great if the dashboard filters by typing in it. The idea is similar to when you start typing in the history window, except that it should filter the visible projects based on what's been typed so far (by sub string match). Then, when either a project is clicked or `Esc` is pressed, it should cancel search mode and show all projects again. Possibly `Enter` could open the first project.

## Environment

- Git Extensions 3.1.0.5877
- Build bb1687f6bd0450d0c1a2bae10a37c65493fc1294
- Git 2.21.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.3752.0
- DPI 96dpi (no scaling)

**Comment by pmiossec** [comment]
I don't think the UX will be good if we do that.

Either we do a fast select (like in the history) or we add a filter textbox
with the focus when form is open (and when ctrl+f).

But we shouldn't do something in between that will feel strange...

**Comment by RussKie** [comment]
I don't think it is feasible... It is probably possible with a search box, like Windows Explorer has it.

Have you considered using dashboard categories? Perhaps this could help in some way

**Comment by KeithSware** [comment]
> context - I'm using version 3.1.0.5877, windows 10, .net v4.6
I agree with the point 'The dashboard tends to become unmanageable when you have a lot of projects', since I'm having that difficulty myself. A sort function would help users to locate repos (sort by name, sort by last updated date, sorted by category group)

Just a thought: 
I use and sometimes rename dashboard categories, BUT, when I want to add a new project to an existing category, I find it difficult to remember which category to add the new project to. To help us out, an improvement would be to allow folks to add a descriptive comment to each category. 

So, Screen space is limited and not all projects will fit into a 27" monitor so it becomes progressively difficult to discover a repo that I want to work with. This gets worse as time passes, as even more git repos are added to the dev box. Also, not all of us have 20-20 vision, so using small fonts does not help.

The number of repos going into the [Recent Repositories] group gets more and more and more, as time goes on. There seems to be no way of filtering projects out of the [Recent Repositories] group, perhaps this is by design?

How about a NEW dashboard group called
Uncategorised  Repositories? 

The [Uncategorised  Repositories] group would contain any project that has not been added into an existing category classification. This would allow me to stop using the [Recent Repositories] group.

Summary
- [ ] new Uncategorised  Repositories group? 
- [ ] Descriptive comment for categories
- [ ] A new sort feature to show all repos, sorted according to alpha name, date last updated, sorted by category group

**Comment by githubkusi** [comment]
I like the initial idea. But if a more conventional approach is desired, just add a text box on top of the recent repositories which allows to filer your recent repos as-you-type, with substring matching
